rosbag2_storage_default_plugins/include/rosbag2_storage_default_plugins/sqlite/sqlite_storage.hpp

```diff
@@ -78,13 +84,17 @@ class ROSBAG2_STORAGE_DEFAULT_PLUGINS_PUBLIC SqliteStorage
     std::shared_ptr<rcutils_uint8_array_t>, rcutils_time_point_value_t, std::string>;
 
   std::shared_ptr<SqliteWrapper> database_;
-  std::string database_name_;
+  std::vector<std::string> database_names_;
   SqliteStatement write_statement_;
   SqliteStatement read_statement_;
   ReadQueryResult message_result_;
   ReadQueryResult::Iterator current_message_row_;
   std::unordered_map<std::string, int> topics_;
   std::vector<rosbag2_storage::TopicMetadata> all_topics_and_types_;
+  uint64_t database_file_counter_ {0};
```

Comment:
You need to update the other attributes too, having half-attributes initialized here and half in the ctor is a recipe for problems.  I.e. `SqliteStatement write_statement_;` -> `SqliteStatement write_statement_ {};` (if it's actually default-constructed)